pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -134,33 +137,33 @@ func (c *CSIMaxVolumeLimitChecker) filterAttachableVolumes(
 			continue
 		}
 
-		driverName, volumeHandle := c.getCSIDriver(pvc)
+		csiNode := nodeInfo.CSINode()
+		driverName, volumeHandle := c.getCSIDriverInfo(csiNode, pvc)
 		// if we can't find driver name or volume handle - we don't count this volume.
 		if driverName == "" || volumeHandle == "" {
 			continue
 		}
+		volumeUniqueName := fmt.Sprintf("%s/%s", driverName, volumeHandle)
 		volumeLimitKey := volumeutil.GetCSIAttachLimitKey(driverName)
-		result[volumeHandle] = volumeLimitKey
+		result[volumeUniqueName] = volumeLimitKey
```

Comment:
So this code will not prevent scheduling of pods to a node which does not have CSI driver installed yet. The KEP states that - for migrated plugins we should immediately stop scheduling of pods to nodes where driver is not installed and for other (non-migrated drivers) after 2 releases we are going to stop scheduling of pods to the node where driver is not installed.

Reply:
I am probably missing something but where did you add that check? AFAICT - we are still not preventing pod scheduling if migration is enabled for a CSI volume and driver is not installed on the node.